### PR TITLE
add test case for ErrWhenBuildListIdx func in db.go and optimize enqueue func in bptree.go and add test case

### PR DESCRIPTION
**Pull request by damotiansheng**
1. add unit test for ErrWhenBuildListIdx  func in db.go
2. optimize enqueue func in bptree.go and add unit test for it

### CODE REVIEW

File: bptree_test.go [hunk 1]
```diff
@@ -544,3 +544,37 @@ func TestBPTree_WriteNode(t *testing.T) {
...
+	// 测试 dequeue() 函数，并验证结果
+	// 预期顺序：1 -> 2 -> 3
+	n := dequeue()
+	if n != node1 {
+		t.Errorf("Dequeue() returned incorrect node. Expected: %v, Got: %v", node1, n)
+	}
+
+	n = dequeue()
+	if n != node2 {
+		t.Errorf("Dequeue() returned incorrect node. Expected: %v, Got: %v", node2, n)
```
Comment (elliotchenzichang) [inline bptree_test.go:568]: In test cases, using assert to check if the value is expected would be better.  Could you help to change it? Thanks!

Comment (elliotchenzichang) [inline bptree_test.go:568]: In test cases, using assert to check if the value is expected would be better.  Could you help to change it? Thanks!

File: bptree_test.go [hunk 2]
```diff
@@ -544,3 +544,37 @@ func TestBPTree_WriteNode(t *testing.T) {
...
+	node1 := &Node{KeysNum: 1}
+	node2 := &Node{KeysNum: 2}
+	node3 := &Node{KeysNum: 3}
+
+	// 测试 enqueue() 函数
+	enqueue(node1)
+	enqueue(node2)
+	enqueue(node3)
+
+	// 测试 dequeue() 函数，并验证结果
```
Comment (elliotchenzichang) [inline bptree_test.go:559]: Could you help translate the comments into English? Thanks!

File: db_test.go [hunk 3]
```diff
@@ -1146,6 +1146,26 @@ func Test_getRecordFromKey(t *testing.T) {
 	}
 }
 
+func TestErrWhenBuildListIdx(t *testing.T) {
+	ts := []struct {
+		err     error
+		want    error
+		notwant error
+	}{
+		{
```
Comment (elliotchenzichang) [inline db_test.go:1155]: Some questions were raised by the review dog. Could you help fix them? Thanks!

**Review by elliotchenzichang — COMMENTED** [review]
Left some comments. Thanks!

**Comment by elliotchenzichang** [comment]
Something failed when building up this PR. And Some unit tests were broken. Could you help to check it? Thanks!

File: bptree_test.go (hunk 1 above)
Comment (damotiansheng) [reply bptree_test.go]: done

Comment (damotiansheng) [reply bptree_test.go]: ok,done

File: bptree_test.go (hunk 2 above)
Comment (damotiansheng) [reply bptree_test.go]: ok

File: db_test.go (hunk 3 above)
Comment (damotiansheng) [reply db_test.go:1156]: has fixed, pls check again, thanks

**Comment by damotiansheng** [comment]
all issues have beed resolved, please help review here

**Review by elliotchenzichang — COMMENTED** [review]
LGTM! Thanks for this contribution.

**Pull request merged**